File: Swift-C++-IR-Comparison/test/arithmetic-operators/Multiplication/Swift/OverflowMul.swift

@main
struct MyMain {

    static func main() {
        // This call would crash with a normal add. 
        // returns -9223372036854775808
        let a = mulWithOverflow(x:Int.max, y:2)
    }

    // The & sign turns an mulition to an mulition without overflow
    static func mulWithOverflow(x: Int, y: Int) -> Int {
        let tmp = x &* y
        return tmp
    }
}

File: AutoTranspiledTests/try-1/OverflowMul.cpp

 #include <iostream>
 
 struct MyMain
 {
   static void main()
   {
     //  This call would crash with a normal mul.
     //  returns -9223372036854775808
     int a = mulWithOverflow(INT_MAX);
   }
 
   // The & sign turns an mul to an mul without overflow
   static int mulWithOverflow(int x)
   {
     int tmp = x &* 2;
     return tmp;
   }
};

int main()
{
  MyMain::main();
  return 0;
}